data-prepper-core/src/test/java/org/opensearch/dataprepper/pipeline/server/PrometheusMetricsHandlerTest.java

```diff
@@ -72,8 +75,22 @@ public void testResponseWritten() throws IOException {
                 .close();
     }
 
+    @Test
+    public void testWhenUseProhibitedHttpMethodThenErrorResponseWritten() throws IOException {
```

Comment:
You can make this a parameterized test. It will allow you to test more scenarios.

It might look like:

```
@ParameterizedTest
@ValueSource(strings = { "DELETE", "PUT" })  /// Put as many as necessary
public void testWhenUseProhibitedHttpMethodThenErrorResponseWritten(String httpMethod) throws IOException {
...
when(exchange.getRequestMethod())
                 .thenReturn(httpMethod);
...
}

```

You can search the code for examples of `ParameterizedTest` and `ValueSource` as well. We do use them in other places.

There are similar tests in this PR that could benefit from this approach.

Reply:
Added parameterized tests in [3ccbc36](https://github.com/opensearch-project/data-prepper/pull/1697/commits/3ccbc36ac5c75a558ec2276ba1701bb66a9483bb)